tests/unit/storage_v3_schema.cpp

```diff
@@ -0,0 +1,310 @@
+// Copyright 2022 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include <gmock/gmock-matchers.h>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <fmt/format.h>
+#include <optional>
+#include <vector>
+
+#include "storage/v2/id_types.hpp"
+#include "storage/v2/property_value.hpp"
+#include "storage/v2/schema_validator.hpp"
+#include "storage/v2/schemas.hpp"
+#include "storage/v2/storage.hpp"
+#include "storage/v2/temporal.hpp"
+
+using namespace memgraph::storage;
+
+using testing::Pair;
+using testing::UnorderedElementsAre;
+
+class SchemaTest : public testing::Test {
+ protected:
+  SchemaTest()
+      : prop1{storage.NameToProperty("prop1")},
+        prop2{storage.NameToProperty("prop2")},
+        label1{storage.NameToLabel("label1")},
+        label2{storage.NameToLabel("label2")},
+        schema_prop1{prop1, memgraph::common::SchemaType::STRING},
+        schema_prop2{prop2, memgraph::common::SchemaType::INT} {}
+
+  memgraph::storage::Storage storage;
```

Comment:
I am wondering whether we need this storage at all. I think it is not necessary. If you want to generate the property/label ids, then I think a NameIdMapper + two helper function are enough. The reason why I think this can be beneficial is it makes harder to introduce any dependency on the storage itself within these tests.